tests/unit_tests/test_async_subtensor.py

```diff
@@ -0,0 +1,291 @@
+from pickle import FALSE
+
+import pytest
+
+from bittensor.core import async_subtensor
+
+
+@pytest.fixture
+def subtensor(mocker):
+    fake_async_substrate = mocker.AsyncMock(
+        autospec=async_subtensor.AsyncSubstrateInterface
+    )
+    mocker.patch.object(
+        async_subtensor, "AsyncSubstrateInterface", return_value=fake_async_substrate
+    )
+    return async_subtensor.AsyncSubtensor()
+
+
+def test_decode_ss58_tuples_in_proposal_vote_data(mocker):
+    """Tests that ProposalVoteData instance instantiation works properly,"""
+    # Preps
+    mocked_decode_account_id = mocker.patch.object(async_subtensor, "decode_account_id")
+    fake_proposal_dict = {
+        "index": "0",
+        "threshold": 1,
+        "ayes": ("0 line", "1 line"),
+        "nays": ("2 line", "3 line"),
+        "end": 123,
+    }
+
+    # Call
+    async_subtensor.ProposalVoteData(fake_proposal_dict)
+
+    # Asserts
+    assert mocked_decode_account_id.call_count == len(fake_proposal_dict["ayes"]) + len(
+        fake_proposal_dict["nays"]
+    )
+    assert mocked_decode_account_id.mock_calls == [
+        mocker.call("0"),
+        mocker.call("1"),
+        mocker.call("2"),
+        mocker.call("3"),
+    ]
+
+
+@pytest.mark.asyncio
+async def test_encode_params(subtensor, mocker):
```

Comment:
I'm not sure what this actually tests. I would think it should fail for invalid types, but it doesn't.

Reply:
As we have already discussed, this tests the actual calls internally. The error case is tested in a subsequent test.